testing/tests/DevExpress.ui.widgets.scheduler/integration.dateNavigator.tests.js

```diff
@@ -24,376 +21,447 @@ QUnit.module("Integration: Date navigator", {
     afterEach: function() {
         fx.off = false;
     }
-});
+};
 
-QUnit.test("Click on the 'next' button should update currentDate", function(assert) {
+QUnit.module("Integration: Date navigator with min and max values", moduleConfig, function() {
+    const DISABLED_CLASS_NAME = "dx-state-disabled";
+    const INIT_CURRENT_DATE = new Date(2017, 4, 25);
 
-    this.createInstance({ currentDate: new Date(2015, 1, 9) });
+    const testNavigatorButtonsState = (assert, instance, cases) => {
+        const $previousButton = $(instance.$element().find(".dx-scheduler-navigator-previous")),
+            $nextButton = $(instance.$element().find(".dx-scheduler-navigator-next"));
 
-    $(this.instance.$element()).find(".dx-scheduler-navigator-next").trigger("dxclick");
+        assert.equal(instance.option("currentDate").valueOf(), INIT_CURRENT_DATE.valueOf(), "currentDate value equal with init currentDate");
+        cases.forEach(testCase => {
+            assert.equal($previousButton.hasClass(DISABLED_CLASS_NAME), testCase.prevButtonDisable);
```

Comment:
Maybe add comments to this asset because using a universal function for testing.